### PR TITLE
update_group_challenge

### LINKED ISSUE #320: Group gets unset when editing challenge
If the challenge has a group set, and someone edits the challenge that does not have access to the group...then the group is unset from the challenge.

The group should not get removed.  It should be a read only field in such cases.

### PR DESCRIPTION
**Pull request by deedee**
- fix #320 
- use group api to get user's group
New token.properties:
@userGroupsApiEndpoint@=<<full url to /groups endpoint >>

### CODE REVIEW

File: token.properties.example [hunk 1]
```diff
@@ -417,6 +417,9 @@
 @member.profile.url.base@=http://tc.cloud.topcoder.com
 
 @memberSearchApiUrl@=https://tc-api.cloud.topcoder.com:8443/v3/members/_suggest/
+@groupMemberSearchApiUrl@=https://cockpit.cloud.topcoder.com/direct/group/member?handle=
+@groupMemberApiUrl@=http://172.18.0.1:8080/v3/groups/%d/members
```
Comment (skyhit) [inline token.properties.example]: @deedee what is this for?

File: token.properties.example [hunk 2]
```diff
@@ -417,6 +417,9 @@
...
 
 @memberSearchApiUrl@=https://tc-api.cloud.topcoder.com:8443/v3/members/_suggest/
+@groupMemberSearchApiUrl@=https://cockpit.cloud.topcoder.com/direct/group/member?handle=
+@groupMemberApiUrl@=http://172.18.0.1:8080/v3/groups/%d/members
 @directChallengeServicesApiUrl@=http://api.topcoder-dev.com/v3/direct/challenges
 @authorizationUrl@=http://api.topcoder-dev.com/v3/authorizations
-@ssoLoginUrl@=https://topcoder-dev.com/login/
\ No newline at end of file
+@ssoLoginUrl@=https://topcoder-dev.com/login/
+@userGroupsApiEndpoint@=http://172.18.0.1:8080/v3/groups
```
Comment (skyhit) [inline token.properties.example]: @deedee can you include the mock api to mock-services repo?

File: token.properties.example (hunk 1 above)
Comment (deedee) [reply token.properties.example]: just update from token.properties.docker

File: token.properties.example (hunk 2 above)
Comment (deedee) [reply token.properties.example]: ok

**Pull request merged**